packages/realm-server/tests/indexing-test.ts

```diff
@@ -556,4 +556,26 @@ module('indexing', function (hooks) {
       'indexed correct number of files',
     );
   });
+
+  test('does not crash when indexing a broken instance', async function (assert) {
+    await realm.write(
+      'broken.json',
+      JSON.stringify({
+        data: {
+          type: 'card',
+          meta: {
+            adoptsFrom: {
+              module: 'https://cardstack.com/base/card-api',
+              name: 'FieldDef',
+            },
+          },
+        },
+      } as LooseSingleCardDocument),
+    ); // this is an example of a card that where loadCard will throw an error
+
+    assert.ok(
```

Comment:
I'm not really happy with this regression test because without the fix, the test runner would just hang after this test and not finish. 

`pnpm test`:
<img width="889" alt="image" src="https://github.com/cardstack/boxel/assets/273660/72056b64-f420-41dc-a714-b66d14b9f984">

Reply:
Is there a way I can test indexing more directly, are there maybe some existing tests for `indexCard` specifically which I can look at?

Reply:
hmmmm, we have realm server tests that live in the host too. that might help you test "closer to the metal" as they say, since the indexing is really host code running in fastboot: 
https://github.com/cardstack/boxel/blob/main/packages/host/tests/integration/realm-test.ts
also here too:
https://github.com/cardstack/boxel/blob/main/packages/host/tests/integration/search-index-test.ts

Reply:
Thanks, the realm-test.ts is a more suitable place for this test. I moved it there in https://github.com/cardstack/boxel/pull/807/commits/679531e483a792222ed6f37ce52dfe4bd0b3cf29